src/Carousel/Carousel.tsx

```diff
@@ -1,64 +1,240 @@
+                    if (xMovement !== swipePos) {
+                        setSwipePos(xMovement - startingPos);
+                    }
+                };
+
+                const endingHandler: VoidFunction = () => {
+                    if (Math.abs(xMovement - startingPos) > parentWidth / 4) {
+                        goToSlide(e, xMovement - startingPos < 0 ? "next" : "prev");
+                    }
+                    setSwipePos(undefined);
+                    document.body.removeEventListener(isTouch ? "touchmove" : "mousemove", movementHandler);
+                    document.body.removeEventListener(isTouch ? "touchend" : "mouseup", endingHandler);
+                };
+
+                document.body.addEventListener(isTouch ? "touchmove" : "mousemove", movementHandler);
+                document.body.addEventListener(isTouch ? "touchend" : "mouseup", endingHandler);
+                isTouch ? props.onTouchStart && props.onTouchStart(e as React.TouchEvent<HTMLDivElement>) : props.onMouseDown && props.onMouseDown(e as React.MouseEvent<HTMLDivElement>);
+            },
+            [swipePos]
+        );
+
+        const interruptionHandler: React.MouseEventHandler<HTMLDivElement> = React.useCallback(
+            (e: React.MouseEvent<HTMLDivElement>): void => {
+                switch (e.type as keyof HTMLElementEventMap) {
+                    case "mouseenter":
+                        interrupted.current = true;
+                        props.onMouseEnter && props.onMouseEnter(e);
+                        break;
+                    case "mouseleave":
+                        interrupted.current = false;
+                        triggerAutoplay();
+                        props.onMouseLeave && props.onMouseLeave(e);
+                        break;
+                }
+            },
+            [triggerAutoplay, props.onMouseEnter, props.onMouseLeave]
+        );
+
+        /** ----- Effects ----- */
+        /** Set a custom ID if there is none */
+        React.useEffect(() => setId(props.id || randomId("carousel-")), [props.id]);
+        /** Set the full size of carousel counting both children and list prop */
+        React.useEffect(() => setSize(list.length + React.Children.count(props.children)), [props.children, list]);
+        /** Sets the default value, if any. Otherwise default to the first item */
+        React.useEffect(() => setActive(props.defaultValue || 0), [props.defaultValue]);
+        /** Set class names */
+        React.useEffect(() => setClassName(classnames("seb", "carousel", { "carousel-fade": transitionStyle === "fade" }, props.className)), [props.className, transitionStyle]);
+        React.useEffect(() => {
+            swipePos && setComingNext(findNewActive(swipePos < 0 ? "next" : "prev"));
+        }, [swipePos]);
+        /** Triggers autoplay if enabled */
+        React.useEffect(() => triggerAutoplay(), [autoplay]);
```

Comment:
I noticed the only dependencies here are `autoplay`, however `trigerAutoplay()` function has more dependencies than autoplay alone (like `autoplaySpeed`). I think this will cause an issue where for example `autoplaySpeed` prop is updated and the trigerAutoplay() callback gets updated but this effect will still not run since it only runs when autoplay changes.

Reply:
`triggerAutoplay` is a `callback` and it caches its own values separately. I'm only triggering the function when `autoplay` prop is changed. I tested it, it seems to be working as intended.

Reply:
Have you tested if you render the component with autoplay = true and autoplaySpeed = 1000 then you change only autoplaySpeed to 5000 right after the initial render. Will the speed actually be 1000 or 5000? From my understanding and what I explained initially it seems like it will be 1000 meaning it will be stale. But I might be wrong

Reply:
it will be for the first round and then changes in the second round. Which is ok for now I guess